### PR TITLE
Download Python versions concurrently in bootstrapping script

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

**Comment by zanieb** [comment]
Doesn't really seem to provide a speed up on CI since it's network IO bound :(

**Comment by zanieb** [comment]
It is quite a bit faster locally though, biasing towards merging.

File: scripts/bootstrap/install.py [hunk 1]
```diff
@@ -173,8 +184,21 @@ def sha256_file(path: Path):
...
-    (THIS_DIR / filename).unlink()
 
-print("Done!")
+if __name__ == "__main__":
+    if INSTALL_DIR.exists():
+        print("Removing existing installations...")
+        shutil.rmtree(INSTALL_DIR)
+
+    # Download in parallel
+    with concurrent.futures.ProcessPoolExecutor(max_workers=len(versions)) as executor:
```
Comment (samypr100) [inline scripts/bootstrap/install.py:194]:
```suggestion
    with concurrent.futures.ProcessPoolExecutor(max_workers=min(os.cpu_count(), len(versions))) as executor:
```

Might be worth to limit by what's available

**Review by konstin — APPROVED** [review]

File: scripts/bootstrap/install.py (hunk 1 above)
Comment (zanieb) [reply scripts/bootstrap/install.py:194]: I don't think overallocating here has much of an affect, I'm just going to leave this as-is for now I hope to move this all to Rust in the near future.

**Pull request merged**